datadog/resource_datadog_synthetics_test_.go

```diff
@@ -216,6 +217,44 @@ func syntheticsTestOptions() *schema.Schema {
 	}
 }
 
+func syntheticsTestStep() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
```

Comment:
This uses a `TypeSet` because a `TypeMap` wouldn't fit (the `params` field is a map), but we need to use a `position` field as steps order is important for browser tests.

Reply:
Can we use `TypeList` to avoid the need for `position` field?

Reply:
Unless I'm mistaken, we can't use a `TypeList` here because one of the field is a `TypeMap` (`params`). So the solution would be to use a `TypeSet`, but with a `position` field since the order is important.

Reply:
I think you're mistaken. Nothing in `TypeList` prevents you from having a map in it.

Reply:
Sorry you're right I was confused. We can indeed use maps in a list, but the issue is the `params` field is a map, and as far as I know we can't have nested maps. So instead of a list of maps with a map inside, I use sets, so we can have a map inside, but we need a `position` field to make sure order is kept.